packages/rum-core/src/performance-monitoring/transaction-service.js

```diff
@@ -178,6 +179,7 @@ class TransactionService {
     if (__DEV__) {
       this._logger.debug('TransactionService.startTransaction', tr)
     }
+    this._config.events.send(ON_TRANSACTION_START, [tr])
```

Comment:
even though I understand the idea behind starting the event here, We have to think about the usecases like this
+ What if the user/us plans to use start and end events to determine the timing of the transaction

We can call the start ASAP but it might have few params not defined but I think that is not a huge issue. 


We are ending the transaction at line https://github.com/elastic/apm-agent-rum-js/pull/279/files#diff-074367311a4235f7774f1d58a437e7b6L157, when the transaction cannot be redefined. Should call onEnd that time as well.

Reply:
We need to run our "fixes" for the transaction first before running the event listeners. we can not run them afterwards, since we might overwrite the user's modifications.

Reply:
The Transaction.onEnd is called when we end the transaction, therefore broadcasting the transaction-end event.